src/const/dataSource.js

```diff
@@ -298,7 +298,7 @@ function mapAttributeToField(section, attribute, index){
     };
 }
 
-function getDataSourceCreationFields(dbmsTypeList, useOnlyVirtual = false) {
+function getDataSourceCreationFields(dbmsTypeList, useOnlyVirtual = false, isStandaloneAndPublished = false) {
```

Comment:
Do you really want such specific name? What's wrong with the suggestion to pass a map of disabled fields?